compiler/luci-interpreter/pal/cmsisnn/PALConv2d.h

```diff
@@ -60,9 +62,82 @@ static inline void ConvPerChannel(const tflite::ConvParams &params, const int32_
 {
   (void)im2col_shape;
   (void)im2col_data;
-  tflite::reference_integer_ops::ConvPerChannel(params, mult, shifts, input_shape, input_data,
-                                                filter_shape, filter_data, bias_shape, bias_data,
-                                                output_shape, output_data);
+
+  cmsis_nn_conv_params conv_params;
+  conv_params.dilation.h = params.dilation_height_factor;
+  conv_params.dilation.w = params.dilation_width_factor;
+
+  if (conv_params.dilation.h == 1 && conv_params.dilation.w == 1)
+  {
+    conv_params.input_offset = params.input_offset;
+    conv_params.output_offset = params.output_offset;
+    conv_params.stride.h = params.stride_height;
+    conv_params.stride.w = params.stride_width;
+    conv_params.padding.h = params.padding_values.height;
+    conv_params.padding.w = params.padding_values.width;
+    conv_params.activation.min = params.quantized_activation_min;
+    conv_params.activation.max = params.quantized_activation_max;
+
+    cmsis_nn_per_channel_quant_params quant_params;
+    quant_params.multiplier = const_cast<int32_t *>(mult);
+    quant_params.shift = const_cast<int32_t *>(shifts);
+
+    TFLITE_DCHECK_LE(conv_params.activation.min, conv_params.activation.max);
+    TFLITE_DCHECK_EQ(input_shape.DimensionsCount(), 4);
+    TFLITE_DCHECK_EQ(filter_shape.DimensionsCount(), 4);
+    TFLITE_DCHECK_EQ(output_shape.DimensionsCount(), 4);
+    const int batch_size = tflite::MatchingDim(input_shape, 0, output_shape, 0);
```

Comment:
Q) Did you copy this from TF repo?
Maybe add TF developers in licence plate?

Reply:
Related tflite code:
https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/kernels/cmsis_nn/conv.cc#L128

Reply:
This is how diff o this code with tflite is looking after remove of comments and reformating:
left is tflite, right is this PR
![image](https://user-images.githubusercontent.com/3631351/141997443-6acdd35d-c99a-4867-a9de-1b54feffb0de.png)

Personally, I think that it would be safer to add TF in license after all.